caffe2/operators/fully_connected_op_gpu.cc

```diff
@@ -2,9 +2,51 @@
 #include "caffe2/operators/fully_connected_op.h"
 
 namespace caffe2 {
+
+template <>
+bool FullyConnectedOp<CUDAContext>::RunOnDevice() {
+  if (Input(0).IsType<float>()) {
+    return DoRunWithType<float,    // X
+                         float,    // W
+                         float,    // B
+                         float,    // Y
+                         float>(); // Math
+  } else if (Input(0).IsType<float16>()) {
+    return DoRunWithType<float16,    // X
+                         float16,    // W
+                         float16,    // B
+                         float16,    // Y
+                         float>(); // Math
+  }
+}
```

Comment:
I assume the nervana change was meant to be caffe2/contrib/nervana/nervana_math_gpu.cc

Changes made.